app/src/ingest_bem_pdfs.py

```diff
@@ -76,15 +80,59 @@ def _parse_pdf(file: BinaryIO) -> list[EnrichedText]:
     for text in grouped_texts:
         text.id = str(uuid.uuid1())
     assert len(set(text.id for text in grouped_texts)) == len(grouped_texts)
+
     return grouped_texts
 
 
-def enrich_texts(file: BinaryIO) -> list[EnrichedText]:
+def enrich_texts(file: BinaryIO, unstructured_json: list[Element]) -> list[EnrichedText]:
+    enrich_text_list = []
     "Placeholder function. Will be implemented for DST-414, probably in a different file."
     outline: list[Heading] = pdf_utils.extract_outline(file)
-    for _heading in outline:
-        pass
-    return []
+    current_header = []
+    current_header_level = 1
+    for element in unstructured_json:
+        if element.category == "Footer" or element.category == "Header":
+            continue
+        if element.category == "Title":
+            header = match_heading(outline, element.text, element.metadata.page_number)
+            if header:
```

Comment:
A few other thoughts on this section:
 - I think "heading" is a better name than "header" here (we're trying to use header/fooder exclusively for the top and bottom of pages), so these would be `current_heading`, etc. (Or maybe `current_headings`, since it's a list)
 - could you separate this out into a function, like get_current_heading(header, current_header) 
 - could we treat current_header_level = 0 when current_header is []? If we refactor the code to handle this, then we don't need the current_header_level variable at all, since the current level will always be `len(current_header)`